app/views/categories/show.html.erb

```diff
@@ -10,6 +10,19 @@
   <%= @category.description %>
 </p>
 
+<p>
+  <strong>Carriers:</strong>
+  <% if @category.carriers.empty? %>
```

Comment:
@m-pereira I understand why you've choose `empty?` method here, but for this use case it is not the optimal relation method to use.

This will perform 2 DB queries if we have carriers
1. Calling `@category.carriers.empty?` will perform `SQL COUNT` query (Rails doesn't memoize empty?)
2. Line 19 will perform `SLECT * FROM carriers...`

Instead it would be better to use `@category.carriers.blank?`. `blank?` is gonna load entire collection(`SELECT * FROM carriers...`) and memoize it...so if there are `carriers` it will not perform 2 DB query but use one from cache/memoization. **It will hit the DB once.**

As a general rule of tumb:

`any?, none?, empty?` - should be used if you only gonna use section of the `ActiveRecord::Relation`, for example: `@category.carriers.first(3).each...`

`present?, blank?` - should be used if `ActiveRecord::Relation` is gonna be used in its entirety, such as this case.

Reply:
Nice! thanks for the tips! I'm going to change it in one minute.